base/stacktraces.jl

```diff
@@ -375,4 +375,15 @@ function from(frame::StackFrame, m::Module)
     return parentmodule(frame) === m
 end
 
+end  # module StackTraces
+
+# Back to Base
+# Implement locking for Compiler.store_dispatch_backtrace
+const dispatch_backtrace = IdDict{Core.CodeInstance,Any}()
+const dispatch_backtrace_lock = ReentrantLock()
```

Comment:
ReentrantLock may be somewhat segfault-y when used in inference itself (since it does not expect to block, it violates the lock hierarchy also). This needs to be in C, since it cannot be running code that can be inferred